sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -264,12 +281,12 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         df,
         sink,
         outputMode,
-        useTempCheckpointLocation = true,
+        useTempCheckpointLocation = isTempCheckpointLocationAvailable,
         trigger = trigger)
     } else {
       val (useTempCheckpointLocation, recoverFromCheckpointLocation) =
         if (source == "console") {
-          (true, false)
+          (isTempCheckpointLocationAvailable, false)
```

Comment:
Yes, you are right. Because if they are different, now you would get the hard-to-understand `PermissionDenied` exception, which gives you no hint about how to solve the problem.